### ISSUE TITLE
Unable to make a metagenotype in PHI-Canto

### ISSUE THREAD

**Issue by CuzickA** [pombase/canto#1797]
Kim (H-K) is planning on giving a live demo of PHI-Canto on Monday/Tuesday. I was just having a run through of an example on PHI-Canto but cannot make a metagenotype. Please could you have a try and see if you encounter the same problem?

I have tried on both the demo and canto sites (logged in as admin and not logged in as admin) using a mixture of pathogen genes and hosts and keep getting this red error message (1st screen shot below).

Upon further investigation it appears that this error occurs when there is a ‘pathogen genotype’ and a ‘wildtype host species’ with no host genes specified (our most common type of expt).

When there is a pathogen genotype and a host genotype the metagenotype is created (2nd screen shot below). This is a rarer type of expt where both a pathogen and host gene are altered. (Note this could not be done using the pathogen genotype page ‘metagenotype management’ button, I had to go back through the ‘meta-genotype management’ link on the summary page.

Is there a problem with the loading of the host species lists?

Best wishes,
Alayne
![image](https://user-images.githubusercontent.com/11753634/54429054-6375df80-4717-11e9-9cfc-283ec2ed571b.png)
![image](https://user-images.githubusercontent.com/11753634/54429079-6ec90b00-4717-11e9-8e2e-2822221727c2.png)

**Comment by kimrutherford** [comment]
I think I've fixed the code.  @jseager7 will need to update the server(s) to see the change.

There were two problems.

The first was that the list of wild type genotypes was empty.  After the fix it has a wild type for each strain:
![wild-type-genotype-list-1](https://user-images.githubusercontent.com/90474/54468989-3b5dad80-47f7-11e9-9b0f-094d584c890b.png)


The second problem was that even with the list working the meta-genotype creation failed.  I think I've fixed that too.

(@jseager7 it was assuming that a function call returned a strain but in fact it was returning a Promise - see ad2be94dae20208c9e0d0f3)

**Comment by jseager7** [comment]
Thanks for your help @kimrutherford. I've now applied this fix to the main server and it seems to be working correctly &ndash; I was able to create a metagenotype using a pathogen genotype and a host with no genes. I haven't applied the fix to the demo server yet, since that server's currently being used for a demonstration at a conference and I don't want to risk causing any interference. I should be able to apply the demo fix once that's over.
